go/billing/api.py

```diff
@@ -247,6 +250,32 @@ def create_transaction_interaction(self, cursor, account_number,
         session_len_credits = MessageCost.calculate_session_length_credit_cost(
             session_len_cost, markup_percent)
 
+        query = """SELECT credit_balance, last_topup_balance
+                   FROM billing_account
+                   WHERE account_number = %(account_number)s"""
+
+        params = {'account_number': account_number}
+        cursor = yield cursor.execute(query, params)
+        result = yield cursor.fetchone()
+
+        if result is None:
+            raise BillingError(
+                "Unable to find billing account %s while checking"
+                " credit balance. Message was %s to/from tag pool %s." % (
+                    account_number, message_direction, tag_pool_name))
+
+        last_topup_balance = result.get('last_topup_balance')
+        credit_balance = result.get('credit_balance')
+
+        def credit_percent():
+            return self._ceil_percent(credit_balance, last_topup_balance)
```

Comment:
Give that we update the value of ``credit_balance``, perhaps we should rather not wrap this in a closure any more and instead just call ``self._ceil_percent`` directly?